### PR TITLE
add Right-to-Left support

### LINKED ISSUE #2762: Right-to-Left language compatibility
As some know, the full Persian interface was translated and released several versions ago.
Since then, our tester found that "some texts are really
messed up in absence of RTL".

The needed changes for Right-To-Left compatibility have been completed, and a PR is forthcoming.

### PR DESCRIPTION
**Pull request by initCCG**
This is regarding issue:
closes #2762

### CODE REVIEW

File: core/src/main/java/bisq/core/locale/LanguageUtil.java [hunk 1]
```diff
@@ -84,6 +84,12 @@
             */
     );
 
+    public static final List<String> rtlLanguagesCodes = Arrays.asList(
```
Comment (ripcurlx) [inline core/src/main/java/bisq/core/locale/LanguageUtil.java]: Can be private

File: desktop/src/main/java/bisq/desktop/main/MainView.java [hunk 2]
```diff
@@ -167,6 +169,8 @@ public MainView(MainViewModel model,
     @Override
     protected void initialize() {
         MainView.rootContainer = root;
+        if (LanguageUtil.isDefaultLanguageRTL())
+	    MainView.rootContainer.setNodeOrientation(NodeOrientation.RIGHT_TO_LEFT);
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/main/MainView.java]: Please apply Bisq coding conventions and style guide: https://github.com/bisq-network/bisq/blob/master/CONTRIBUTING.md#style-and-coding-conventions

File: desktop/src/main/java/bisq/desktop/bisq.css [hunk 3]
```diff
@@ -165,6 +165,9 @@ bg color of non edit textFields: fafafa
     -fx-selection-bar-non-focused: -fx-selection-bar;
 
     -fx-font-size: 13;
+}
+
+.root:dir(ltr){
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/bisq.css:170]: Why was this necessary?

**Review by ripcurlx — CHANGES_REQUESTED** [review]
NACK - based on my inline comments. These required changes could be easily implemented, but I recognized that there are some issues with the tabs and toggle buttons we are using.

Bildschirmfoto 2019-04-25 um 14 28 45
Tab labels are squashed and hardly readable and the toggle is slightly misplaced and squashed as well. I'll see if I update the library if the problems were fixed already within newer versions.

File: desktop/src/main/java/bisq/desktop/bisq.css (hunk 3 above)
Comment (TechMiX) [reply desktop/src/main/java/bisq/desktop/bisq.css:170]: for some reason, setting `-fx-font-family` in the `.root` class was breaking the Bi-directional system in Javafx and thus texts with mixed language characters were messed up. I limited it to left-to-right direction only.

**Comment by ripcurlx** [comment]
@TechMix Maybe you could have a look if you don't set the Node Orientation for Tab and the AutoTooltipSlideToggleButton that the remaining UI problems are gone as well? https://github.com/bisq-network/bisq/pull/2774#pullrequestreview-230612058

**Comment by TechMiX** [comment]
@ripcurlx Actually I can't test it. Because I'm not experiencing it on my Ubuntu machine:
scrnshot
Could you please test it without applying this PR on your machine to see if it's related to font?

**Comment by ripcurlx** [comment]
@TechMiX hmm - that's weird. I tried it already on my machine and I don't see the squashing of the text in the tabs  and the tab misplacement in current master. Still there seems to be an issue with the IBM Sans font in the main buttons. 
Bildschirmfoto 2019-04-25 um 14 33 22

**Comment by ripcurlx** [comment]
@TechMiX I quickly tried the latest version of the UI library we are using, but it didn't solve the problem. Maybe it has to do with some customization from our side. I'll have a look at it later to solve it also for macOS (no idea why it behaves differently on linux)

**Comment by TechMiX** [comment]
@ripcurlx OK. Should I make any other changes for this PR?

**Comment by ripcurlx** [comment]
> @ripcurlx OK. Should I make any other changes for this PR?

It would be great if you could try to be able to test it on macOS and fix the remaining problem. I can have a look at it myself, but I have to focus on anti-scam features right now, before being able to do so.

**Comment by TechMiX** [comment]
> > @ripcurlx OK. Should I make any other changes for this PR?
> 
> It would be great if you could try to be able to test it on macOS and fix the remaining problem. I can have a look at it myself, but I have to focus on anti-scam features right now, before being able to do so.

Unfortunately, I don't have access to a macOS machine. I've tested it on Ubuntu and Windows.

**Review by ripcurlx — APPROVED** [review]

**Comment by ripcurlx** [comment]
@TechMiX Just rebased it with master and tried it again and it seems to work now. No idea why it wasn't working before as we didn't update the UI library since then. Sorry for the long merge delay 😬 . But we'll never forget a PR 😉 
Bildschirmfoto 2019-09-19 um 16 02 46
